### DISCUSSION TITLE
Sweeter progress overview

### DISCUSSION THREAD

**Discussion by Duke-M-commits** [ReactiveDrop/reactivedrop_translations#512] [Ideas]
As of now the table in progress.md has some downsides:
* Table is cut off because of long contents per cell (the git webview is probably the most seen visual representation for that file)
* Percentages with 3 decimal places are a bit too detailed to be nice to read (the file is meant to be read by humans, it wont get reread by a machine)
* Fully translated languages are included in the table which rises the width further (but there isn't a breakdown information in these to be found by the reading person)

I sketched a shorter and easier to read document:
* shortened the language-columns (first 3 characters of language string) and added tooltips with full names on hover
* removed percentages in the whole (that's something for a nice comparison graph)
* removed fully translated language columns
* added a list of fully translated languages at the top
* removed non curated language columns
* added a list of non curated languages at the bottom
* linked files to their directories

I created a mockup [here](https://github.com/DukeDukeEm/rd_translation_german/blob/progress_sketch/PROGRESS.md)
How about that?
And why? I think a good representation of the progress could motivate new visitors to contribute.

Disclaimer: I can't write GO, so I can only provide my input, somebody else would have to change `update_translations.go`

Before:
![image](https://user-images.githubusercontent.com/69652463/199941140-1f111d9c-89f4-45a2-b9d3-b5cc2cc8c52d.png)
After:
![image](https://user-images.githubusercontent.com/69652463/199940947-8ab4fbf9-a649-416c-ab93-74f92830e6bc.png)

**Comment by Blueberryy** [comment]
> As of now the table in progress.md has some downsides:
> * Table is cut off because of long contents per cell (the git webview is probably the most seen visual representation for that file)
> * Percentages with 3 decimal places are a bit too detailed to be nice to read (the file is meant to be read by humans, it wont get reread by a machine)
> * Fully translated languages are included in the table which rises the width further (but there isn't a breakdown information in these to be found by the reading person)
> 
> I sketched a shorter and easier to read document:
> * shortened the language-columns (first 3 characters of language string)
> * removed percentages in the whole (that's something for a nice comparison graph)
> * removed fully translated columns
> * added a list of fully translated languages at the top
> 
> I created a mockup [here](https://github.com/DukeDukeEm/rd_translation_german/blob/progress_sketch/PROGRESS.md)
> How about that?
> 
> Disclaimer: I can't write GO, so I can only provide my input, somebody else would have to change `update_translations.go`
> 
> Before:
> ![image](https://user-images.githubusercontent.com/69652463/199699253-26479b7d-6d98-4101-a082-85311105a391.png)
> After:
> ![image](https://user-images.githubusercontent.com/69652463/199699350-dcbc43e4-8b15-4163-acdc-b72be89e6a7a.png)
> 

It would be cool to add "Very Outdated Languages" at the bottom of that list for languages that supported by game but never updated by someone since original Alien Swarm game like you did for "Fully Translated Languages"

**Comment by Duke-M-commits** [comment]
Thank you, that's a very good idea, I updated my [mockup](https://github.com/DukeDukeEm/rd_translation_german/blob/progress_sketch/PROGRESS.md)

**Comment by Duke-M-commits** [comment]
I added links to the listed files (to be exact, their directories). 
While doing this I realized there seem to be some new files in the directories which aren't part of the statistic.
* ../community/app_items
* ../community/eula
* ../community/item_schema
* ../community/statsweb
* ../community/workshop_description
* ../community/workshop_tags
* ../credits/asw_credits
* ../credits/biogen_corporation_credits
* ../credits/credits_acc32_official
* ../credits/credits_namhumanum
* ../credits/deathmatch_credits
* ../credits/rd-LanasEscapeCredits
* ../credits/rd-area9800Credits
* ../credits/rd-ocs-credits
* ../credits/rd-tarnor_credits
* ../credits/rd_bonus_missions_credits
* ../credits/rd_paranoia_credits
* ../credits/rd_research7_credits
* ../credits/tilarus5_credits
* ../rich_presence/563560_loc
* ../store_page/content_warning
* ../store_page/storepage_81731
* ../addons/ExampleAddon/resource/CustomCampaignCredits

PS: while making this list I realized, it could be that the mail and the news folder aren't in there either

**Comment by Duke-M-commits** [comment]
This idea turned into an [issue](https://github.com/ReactiveDrop/reactivedrop_translations/issues/520) and has been resolved in a [beautiful and useful report](https://github.com/ReactiveDrop/reactivedrop_translations/blob/master/PROGRESS.md).

**Comment by Duke-M-commits** [comment]
How about adding a line about the number of translateable lines?
Like, above the table `There are currently <number> translateable lines.`
I'm not sure if you would need that but I wrote an regex to find out for myself:
https://github.com/ReactiveDrop/reactivedrop_translations/wiki/How-to-translate-using-vscode#count-all-translateable-lines
